src/sdltiles.cpp

```diff
@@ -2866,17 +2866,16 @@ static void CheckMessages()
                         break;
 #endif
                     case SDL_WINDOWEVENT_SHOWN:
-                    case SDL_WINDOWEVENT_EXPOSED:
                     case SDL_WINDOWEVENT_MINIMIZED:
-                        break;
                     case SDL_WINDOWEVENT_FOCUS_GAINED:
+                        break;
+                    case SDL_WINDOWEVENT_EXPOSED:
                         // Main menu redraw
                         reinitialize_framebuffer();
-                        // TODO: redraw all game menus if they are open
+                        ui_manager::redraw_all();
```

Comment:
Please test it in fullscreen dx9 mode. I remember in fullscreen mode when the game window was maximized after alt-tabbing, an event appeared - SDL_WINDOWEVENT_FOCUS_GAINED.

Reply:
I've tested in fullscreen with direct3d and it was refreshing properly. I didn't see a dx9 mode though, so I'm not sure if that's what you mean.

Reply:
Ok.